Feature/populate level 1

tedgag opened the pull request:
Added enemies, interactables, weapons, etc. to level 1.
Known bugs:
- Bullets behind walls
- Enemies soaking bullets even after death
- Boss immune to explosions

WillTarte commented:
Overall, excellent work but there's a few issues:
1. Dialogues are bugged: The boxes appear, but there's no character picture and no text (manually enabling them works tho)
2. For some reason assault rifle has huge laser bullets (will create a bug ticket for this)
3. Melee emu follow range seems low (should be easy to increase it a bit)
4. Can't see interactables prompt (this is a rendering layers thing, will create a ticket for this)
5. Infinite lives due to checkpoints (never gameover) + respawn with no weapons (would be good that checkpoints respawn the player with inventory + hp)
6. Boss is not doing anything (hp bar triggered and takes damage, but doesnt move/attack)

tedgag commented:
> Overall, excellent work but there's a few issues:
> 
> 1. Dialogues are bugged: The boxes appear, but there's no character picture and no text (manually enabling them works tho)
> 2. For some reason assault rifle has huge laser bullets (will create a bug ticket for this)
> 3. Melee emu follow range seems low (should be easy to increase it a bit)
> 4. Can't see interactables prompt (this is a rendering layers thing, will create a ticket for this)
> 5. Infinite lives due to checkpoints (never gameover) + respawn with no weapons (would be good that checkpoints respawn the player with inventory + hp)
> 6. Boss is not doing anything (hp bar triggered and takes damage, but doesnt move/attack)

1. This bug should be fixed but for some reason some of them still don't work, will probably create a ticket so someone lese can look at it
3. I think it's fine as is
5. Probably ok, we don't have time to add lives and it will be easier to test the game. I've added a weapon near all checkpoints.
6. Fixed
